src/main/java/org/launchcode/techjobs/persistent/controllers/HomeController.java

```diff
@@ -26,27 +42,36 @@ public String index(Model model) {
     @GetMapping("add")
     public String displayAddJobForm(Model model) {
         model.addAttribute("title", "Add Job");
+        model.addAttribute("employers", employerRepository.findAll());
+        model.addAttribute("skills", skillRepository.findAll());
         model.addAttribute(new Job());
         return "add";
     }
 
     @PostMapping("add")
     public String processAddJobForm(@ModelAttribute @Valid Job newJob,
-                                       Errors errors, Model model, @RequestParam int employerId, @RequestParam List<Integer> skills) {
+                                    Errors errors, Model model, @RequestParam int employerId, @RequestParam List<Integer> skills) {
 
         if (errors.hasErrors()) {
             model.addAttribute("title", "Add Job");
             return "add";
         }
+        Optional<Employer> result = employerRepository.findById(employerId);
+        if (!result.isEmpty()) {
+            Employer employer = result.get();
+            newJob.setEmployer(employer);
+            jobRepository.save(newJob);
```

Comment:
Here, you're saving the new job before you've set the skills. It runs this line of code before it runs line 67, where it sets the skills.I'm curious to see if the data created was created with skills or not? It shouldn't based on what I'm seeing here.